patches/net/minecraft/world/level/material/Fluid.java.patch

```diff
@@ -1,5 +1,13 @@
 --- a/net/minecraft/world/level/material/Fluid.java
 +++ b/net/minecraft/world/level/material/Fluid.java
+@@ -35,6 +_,7 @@
+         this.createFluidStateDefinition(builder);
+         this.stateDefinition = builder.create(Fluid::defaultFluidState, FluidState::new);
+         this.registerDefaultState(this.stateDefinition.any());
++        this.defaultResource = net.neoforged.neoforge.common.util.Lazy.of(() -> net.neoforged.neoforge.transfer.resources.FluidResource.createDefaultInstance(this));
```

Comment:
See my comment on the Item class, for what I think of this mechanic.

TLDR: Over-engineered and not needed.